presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -190,8 +192,8 @@ public static OrcPageSource createOrcPageSource(
         AggregatedMemoryContext systemMemoryUsage = newSimpleAggregatedMemoryContext();
         try {
             OrcReader reader = new OrcReader(orcDataSource, orcEncoding, maxMergeDistance, maxBufferSize, tinyStripeThreshold, maxReadBlockSize);
-
-            List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useOrcColumnNames, reader, path);
+            boolean useColumnNames = shouldUseColumnNames(schema, useOrcColumnNames);
+            List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useColumnNames, reader, path);
```

Comment:
This uses the column names of the file to match against the partition columns. This is ORC's own tolerance for re-ordering of columns. It won't cover table vs. partition changes.